### ISSUE TITLE
textFont() and textSize() issues using system font after custom font

### ISSUE THREAD

**Issue by dhowe** [processing/p5.js#930]
Neither function (textFont and textSize) works correctly if using a system font AFTER a custom font in a single sketch:

``` js
    function setup()
    {
      createCanvas(600, 300);
      loadFont("Resagokr.otf", function(font)
      {
        background(200);
        textFont(font, 100);
        textAlign(CENTER);
        text("Custom", width / 2, 100);
        textFont("Times");
        textSize(100);
        text("Times", width / 2, 200);
      });
    }
```

ACTUAL:
![screen shot 2015-09-20 at 4 36 29 pm](https://cloud.githubusercontent.com/assets/737638/9979882/c61b0e1e-5fb5-11e5-88c3-2912c913ff32.png)

EXPECTED:
![screen shot 2015-09-20 at 4 36 29 pm](https://cloud.githubusercontent.com/assets/737638/9979891/1fb276b0-5fb6-11e5-81b4-75aed1b1c5e6.png)

**Comment by lmccart** [comment]
when I try the code above I see this (using different custom font, AvenirNextLTPro-Demi.otf)

<img width="620" alt="screen shot 2015-09-20 at 3 57 16 pm" src="https://cloud.githubusercontent.com/assets/191056/9982614/5deb34a4-5fb0-11e5-8ded-a8c9798bb6bf.png">

**Comment by dhowe** [comment]
Rather strange... seems that your font is the only one (I've tried 5 others now) for which it works correctly thus far. Can you try with [this one](http://rednoise.org/imc/uploads/Site/Resagokr.otf)?

**Comment by lmccart** [comment]
yup, that one doesn't work. I tested a few others. it seems sometimes the scale works but the styling (italic, bold, etc) is affected.
SourceSansPro-Italic.ttf
<img width="618" alt="screen shot 2015-09-21 at 7 42 46 am" src="https://cloud.githubusercontent.com/assets/191056/9991046/91057f00-6034-11e5-841d-906e633b010b.png">
SourceSansPro-Bold.ttf
<img width="619" alt="screen shot 2015-09-21 at 7 43 05 am" src="https://cloud.githubusercontent.com/assets/191056/9991048/910a1cae-6034-11e5-860b-710e8fd195d8.png">
SourceSansPro-Regular.otf
<img width="617" alt="screen shot 2015-09-21 at 7 43 26 am" src="https://cloud.githubusercontent.com/assets/191056/9991047/9108631e-6034-11e5-948c-5160ace8fc36.png">

**Comment by dhowe** [comment]
Strange... shall I check in a test case under examples/p5.Font/bug930 ?

**Comment by bleepsandblops** [comment]
I have the exact same issue, it's driving me crazy... the custom font I'm using is Euclid Flex.

**Comment by dhowe** [comment]
please include a link to the font if you don't mind

**Comment by bleepsandblops** [comment]
I can't I'm afraid as it's under license... this is where it's from https://www.swisstypefaces.com/fonts/euclid-flex/

In the meantime I've browsed the closed issues and one was saying to load it via css (should have thought about this before!).

**Comment by tstefanich** [comment]
This is a really bizarre issue... 

For example with this code in the _applyTextProperties Function 

```
p5.Renderer2D.prototype._applyTextProperties = function() {

  var font, p = this._pInst;

  p._setProperty('_textAscent', null);
  p._setProperty('_textDescent', null);

  font = p._textFont;

  if (p._isOpenType()) {=
    font = p._textFont.font.familyName;
    p._setProperty('_textStyle', p._textFont.font.styleName);
  }
  console.log(p._textStyle + ' ' + p._textSize + 'px ' + font);
  this.drawingContext.font = p._textStyle + ' ' + p._textSize + 'px ' + font;
  console.log(this.drawingContext.font);

  return p;
};
```

The two console logs that return two totally different results even though they should be the same 
![Image of console log](https://photos-5.dropbox.com/t/2/AABRXfPLrlpQ_imASAiHyxS-Wj3LHyLqr-cNfgV8YCzGnQ/12/3905701/png/32x32/3/1486299600/0/2/Screenshot%202017-02-05%2000.09.49.png/EOn09wIYrNc-IAcoBw/DsJY0_tJMNQ39Fw5dnxkHQqbu4JRhQm5DHjohB2FQFw?dl=0&size=2048x1536&size_mode=3)

Some how the drawingContext.font get reset... not entirely sure how considering the two console lines are right next to each other. But I think somehow the _applyDefaults() is getting called somewhere becuase if I comment out the line with font defaults 
```
this.drawingContext.font = 'normal 12px sans-serif';
```
it defaults to 10px san-serif instead...

Another weird part is I am unsure where the P._textStyle "regular" is coming from...

**Comment by tstefanich** [comment]
This might be canvas problem and not p5 ... http://stackoverflow.com/questions/3349947/html-5-canvas-font-being-ignored

Also that "Regular" _textStyle that is showing up will break font rendering for sure....

**Comment by seymar** [comment]
I'm also still having this problem, it's really annoying.

**Comment by dhowe** [comment]
I believe this was fixed with https://github.com/processing/p5.js/pull/2106
@lmccart perhaps you can verify and close

**Comment by lmccart** [comment]
confirming this looks good across browsers with fix from #2106, thanks @dhowe!
